src/network/wait-online/link.c

```diff
@@ -246,5 +246,11 @@ int link_update_monitor(Link *l) {
         else
                 free_and_replace(l->state, state);
 
+        r = sd_network_link_get_dns_default_route(l->ifindex);
```

Comment:
this information should be provided by resolved I am sure. (see comment on earlier commit)